src/main/java/org/broadinstitute/hellbender/tools/copynumber/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.ChunkedCopyNumberPosteriorCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.SimpleIntervalCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.records.LocatableCopyNumberPosteriorDistribution;
+import org.broadinstitute.hellbender.tools.copynumber.gcnv.GermlineCNVPostProcessingEngine;
+import org.broadinstitute.hellbender.tools.copynumber.gcnv.GermlineCNVNamingConstants;
+import org.broadinstitute.hellbender.tools.copynumber.gcnv.IntegerCopyNumberStateCollection;
+import org.broadinstitute.hellbender.utils.text.XReadLines;
+import org.broadinstitute.hellbender.utils.tsv.TableUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Create a VCF given the output of {@link GermlineCNVCaller}.
+ *
+ * <p> This tool takes a list of {@link GermlineCNVCaller} call directories containing posterior records for
+ * different copy number states and outputs a VCF for a specified sample. Note that the list should be sorted
+ * according order of the intervals in corresponding chunks. The VCF will contain an ALT allele for every
+ * non-reference copy number state specified in the posterior files. The reference allele corresponds to copy number 2.
+ * </p>
+ *
+ * <h3>Inputs</h3>
+ * <ul>
+ *     <li>A list of paths to {@link GermlineCNVCaller} chunked calls directories</li>
+ *     <li>Name of the sample directory (it must be the same across all chunks)</li>
+ * </ul>
+ *
+ * <h3>Output</h3>
+ * <ul>
+ *     <li>A VCF file with CNV calls</li>
+ * </ul>
+ *
+ * <h3>Usage example</h3>
+ *
+ * <pre>
+ *   gatk GenerateVCFFromPosteriors \
+ *     --called-chunk-directory path/to/chunk_1
+ *     --called-chunk-directory path/to/chunk_2
+ *     --sample-name-directory SAMPLE_1
+ *     --output output.vcf
+ * </pre>
+ */
+@CommandLineProgramProperties(
+        summary = "Create a VCF given the output of GermlineCNVCaller.",
+        oneLineSummary = "Create a VCF given the output of GermlineCNVCaller.",
+        programGroup = CopyNumberProgramGroup.class
+)
+public final class GenerateVCFFromPosteriors extends GATKTool {
+    private static final Logger logger = LogManager.getLogger(GenerateVCFFromPosteriors.class);
+
+    public static final String POSTERIOR_CALL_DIRECTORY_FULL_NAME = "chunk-path";
```

Comment:
Two things: 1) let's make the variable name reflect the kebab-case argument for both of these, and 2) for whatever reason, we typically use `LONG_NAME` instead of `FULL_NAME`.  So, `CHUNK_PATH_LONG_NAME` and `SAMPLE_DIRECTORY_LONG_NAME`.

Reply:
Done